src/sentry/snuba/metrics/naming_layer/mapping.py

```diff
@@ -0,0 +1,71 @@
+__all__ = ("create_name_mapping_layers", "get_mri", "get_reverse_mri")
+
+
+from enum import Enum
+from typing import Dict, Union, cast
+
+from sentry.snuba.metrics.naming_layer.mri import SessionMRI, TransactionMRI
+from sentry.snuba.metrics.naming_layer.public import SessionMetricKey, TransactionMetricKey
+from sentry.snuba.sessions_v2 import InvalidField
+
+
+def create_name_mapping_layers() -> None:
+    # ToDo(ahmed): Hack this out once the FE changes their mappings
+    # Backwards Compat
+    NAMING_MAPPING_LAYER.update(
+        {
+            # Session
+            "sentry.sessions.session": SessionMRI.SESSION,
+            "sentry.sessions.user": SessionMRI.USER,
+            "sentry.sessions.session.duration": SessionMRI.RAW_DURATION,
+            "sentry.sessions.session.error": SessionMRI.ERROR,
+        }
+    )
+
+    for (MerticKey, MRI) in (
+        (SessionMetricKey, SessionMRI),
+        (TransactionMetricKey, TransactionMRI),
+    ):
+        # Adds new names at the end, so that when the reverse mapping is created
+        for metric_key in MerticKey:
+            try:
+                NAMING_MAPPING_LAYER[metric_key.value] = MRI[metric_key.name]
+            except KeyError:
+                continue
+
+    REVERSE_NAME_MAPPING_LAYER.update({v.value: k for k, v in NAMING_MAPPING_LAYER.items()})
+
+
+NAMING_MAPPING_LAYER: Dict[str, Enum] = {}
+REVERSE_NAME_MAPPING_LAYER: Dict[str, str] = {}
```

Comment:
While reviewing I went back to these two dicts repeatedly to check which one maps in which direction. Should we maybe rename them to `NAME_TO_MRI` and `MRI_TO_NAME` or something similar?

Reply:
Agreed